lib/types/SSITypesBuilder.ts

```diff
@@ -46,14 +46,53 @@ export class SSITypesBuilder {
   private static mapExternalVerifiableCredentialToInternal(externalCredential: IVerifiableCredential) {
     if (externalCredential.vc && externalCredential.iss) {
       const vc: InternalVerifiableCredential = new InternalVerifiableCredentialJwt();
-      return Object.assign(vc, externalCredential);
+      const result = Object.assign(vc, externalCredential);
+      return SSITypesBuilder.setJWTAdditionalFields(result);
     } else if (externalCredential.credentialSubject && externalCredential.id) {
       const vc: InternalVerifiableCredential = new InternalVerifiableCredentialJsonLD();
       return Object.assign(vc, externalCredential);
     }
     throw 'VerifiableCredential structure is incorrect.';
   }
 
+  private static setJWTAdditionalFields(result: InternalVerifiableCredentialJwt & IJwtCredential & IHasProof) {
+    if (result.exp) {
+      const expDate = result.getBaseCredential().credentialSubject?.expirationDate;
+      if (expDate && expDate !== result.exp) {
+        throw new Error('Inconsistent expiration dates');
+      }
+      const exp = result.exp.match(/^\d+$/) ? parseInt(result.exp) : result.exp;
+      result.getBaseCredential().credentialSubject.expirationDate = new Date(exp).toISOString();
+    }
+    if (result.iss) {
+      const issuer = result.getBaseCredential().issuer;
+      if (issuer && issuer !== result.iss) {
+        throw new Error('Inconsistent issuers');
+      }
+      result.getBaseCredential().issuer = result.iss;
+    }
+    if (result.nbf) {
+      if (result.getBaseCredential().issuanceDate !== result.nbf) {
+        throw new Error('Inconsistent issuance dates');
+      }
+      const nbf = result.nbf.match(/^\d+$/) ? parseInt(result.nbf) : result.nbf;
+      result.getBaseCredential().issuanceDate = new Date(nbf).toISOString();
+    }
+    if (result.sub) {
+      if (result.getBaseCredential().credentialSubject?.id !== result.sub) {
```

Comment:
This would not work when no credentialSubject.id was provided, as undefined !== result.sub

Reply:
Fixed!